src/js/Maze.js

```diff
@@ -51,6 +51,42 @@ class Maze {
             return index ===cells.length ? false : true;
         }
     }
+
+    /**
+     * Calculate distance of all maze cells from cell at given index,
+     * then use color to visualize distance.
+     *
+     * @param {number} cellIdx - Index of selected start cell.
+     */
+    colorByDistanceFromCell(cellIdx) {
+        const cells = this.getCells();
+        cells.forEach((cell) => cell.distance = null);
+
+        const startCell = cells[cellIdx];
+        startCell.distance = 0;
+
+        let maxDist = 0;
+        let frontierCells = [startCell];
+
+        while (frontierCells.length) {
+            const newFrontier = [];
+            frontierCells.forEach((cell) => {
+                const accessibleNeighbors = cell.getAccessibleNeighbors();
+                const unvisitedAccessible = accessibleNeighbors.filter((neighbor) => neighbor.distance === null);
```

Comment:
The fact that we have methods with the word "visited", which means a totally different thing, on our cell, makes me wonder if we need to change those method names. I feel like the constant re-use of the word "visited" with different meanings would be super confusing for someone jumping into the codebase. 🤔 Definitely seems appropriate here though!!